lib/web_ui/lib/src/engine/pointer_binding.dart

```diff
@@ -156,20 +160,259 @@ class PointerBinding {
+
+    if (!isDebouncing) {
+      // There's no pending queue of pointer events that are being debounced. It
+      // is a standalone click event. Unless pointer down/up were flushed
+      // recently and if the node is currently listening to event, forward to
+      // the framework.
+      if (isListening && _shouldSendClickEventToFramework(click)) {
+        _sendSemanticsTapToFramework(click, semanticsNodeId);
+      }
+      return;
+    }
+
+    if (isListening) {
+      // There's a pending queue of pointer events. Prefer sending the tap action
+      // instead of pointer events, because the pointer events may not land on the
+      // combined semantic node and miss the click/tap.
+      final DebounceState state = _state!;
+      _state = null;
+      state.timer.cancel();
+      _sendSemanticsTapToFramework(click, semanticsNodeId);
+    } else {
+      // The semantic node is not listening to taps. Flush the pointer events
+      // for the framework to figure out what to do with them. It's possible
+      // the framework is interested in gestures other than taps.
+      _flush();
+    }
+  }
+
+  void _sendSemanticsTapToFramework(DomEvent click, int semanticsNodeId) {
+    // Tappable nodes can be nested inside other tappable nodes. If a click
+    // lands on an inner element and is allowed to propagate, it will also
+    // land on the ancestor tappable, leading to both the descendant and the
+    // ancestor sending SemanticsAction.tap to the framework, creating a double
+    // tap/click, which is wrong. More details:
+    //
+    // https://github.com/flutter/flutter/issues/134842
+    click.stopPropagation();
+
+    EnginePlatformDispatcher.instance.invokeOnSemanticsAction(
+        semanticsNodeId, ui.SemanticsAction.tap, null);
+  }
+
+  void _startDebouncing(DomEvent event, List<ui.PointerData> data) {
+    assert(
+      _state == null,
+      'Cannot start debouncing. Already debouncing.'
+    );
+    assert(
+      event.type == 'pointerdown',
+      'Click debouncing must begin with a pointerdown'
+    );
+
+    final DomEventTarget? target = event.target;
+    if (target is DomElement && target.hasAttribute('flt-tappable')) {
+      _state = (
+        target: target,
+        // The 200ms duration was chosen empirically by testing tapping, mouse
+        // clicking, trackpad tapping and clicking, as well as the following
+        // screen readers: TalkBack on Android, VoiceOver on macOS, Narrator/
+        // NVDA/JAWS on Windows. 200ms seemed to hit the sweet spot by
+        // satisfying the following:
+        //   * It was short enough that delaying the `pointerdown` still allowed
+        //     drag gestures to begin reasonably soon (e.g. scrolling).
+        //   * It was long enough to register taps and clicks.
+        //   * It was successful at detecting taps generated by all tested
+        //     screen readers.
+        timer: Timer(const Duration(milliseconds: 200), _onTimerExpired),
```

Comment:
This is not working for long taps (issue https://github.com/flutter/flutter/issues/147050). 

In long tap case, it is possible that `pointerup` event is fired after 200 milliseconds since `pointerdown`, and  as a result `isDebounce` will be `false` when `click` event is received, leading to duplication. 

In the previous solution (that is still exists in parallel), timer was added after `pointerup` (and others) as well, so it is probably should fix the issue
https://github.com/flutter/engine/blob/e5215e6854bc61cf8d5bae48715d73293bc1f91c/lib/web_ui/lib/src/engine/semantics/semantics.dart#L2232-L2250
https://github.com/flutter/engine/blob/e5215e6854bc61cf8d5bae48715d73293bc1f91c/lib/web_ui/lib/src/engine/semantics/semantics.dart#L2184-L2195

Reply:
This is correct, and it's on my radar to fix. This method was enough for the screen reader case. However, we're trying to make it so semantics can be always enabled, and therefore it must also work for all interactions without the screen reader. This code needs a revision.